WordPress/src/test/java/org/wordpress/android/ui/jetpack/scan/details/usecases/IgnoreThreatUseCaseTest.kt

```diff
@@ -0,0 +1,64 @@
+package org.wordpress.android.ui.jetpack.scan.details.usecases
+
+import com.nhaarman.mockitokotlin2.any
+import com.nhaarman.mockitokotlin2.whenever
+import kotlinx.coroutines.InternalCoroutinesApi
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.Before
+import org.junit.Test
+import org.mockito.Mock
+import org.wordpress.android.BaseUnitTest
+import org.wordpress.android.TEST_DISPATCHER
+import org.wordpress.android.fluxc.action.ScanAction.IGNORE_THREAT
+import org.wordpress.android.fluxc.store.ScanStore
+import org.wordpress.android.fluxc.store.ScanStore.IgnoreThreatError
+import org.wordpress.android.fluxc.store.ScanStore.IgnoreThreatErrorType.INVALID_RESPONSE
+import org.wordpress.android.fluxc.store.ScanStore.OnIgnoreThreatStarted
+import org.wordpress.android.test
+import org.wordpress.android.ui.jetpack.scan.details.usecases.IgnoreThreatUseCase.IgnoreThreatState.Failure
+import org.wordpress.android.ui.jetpack.scan.details.usecases.IgnoreThreatUseCase.IgnoreThreatState.Success
+import org.wordpress.android.util.NetworkUtilsWrapper
+
+@InternalCoroutinesApi
+class IgnoreThreatUseCaseTest : BaseUnitTest() {
+    private lateinit var useCase: IgnoreThreatUseCase
+    @Mock lateinit var networkUtilsWrapper: NetworkUtilsWrapper
+    @Mock lateinit var scanStore: ScanStore
+
+    private val fakeSiteId = 1L
+    private val fakeThreatId = 11L
+
+    @Before
+    fun setup() = test {
+        useCase = IgnoreThreatUseCase(networkUtilsWrapper, scanStore, TEST_DISPATCHER)
+        whenever(networkUtilsWrapper.isNetworkAvailable()).thenReturn(true)
+        whenever(scanStore.ignoreThreat(any())).thenReturn(OnIgnoreThreatStarted(IGNORE_THREAT))
```

Comment:
Minor (🔍): Consider moving this `scanStore` mock setup to the last test and have it there explicitly. Below my reasoning:
1) This mock setup and the `networkUtilsWrapper` mock setup are not the same. In the network care we might want to apply the default to be `true`, which will be applicable to all tests, and have it only being overitten by a small number of tests (in that case just one). This is very valid in order to avoid duplication and increase readability.
2) However, for the `scanStore` case, by adding the default behavior to the `success` path, it also seems that we want to apply that to all test, but this is not true as this mock setup is only required for one test, thus it can make things ambiguous.
3) By having the `scanStore` mock setup explicitly in the last test, it can be compared easily with the above `failure` path and thus increase test coherence and consistency.

PS: Please feel free to ignore me. I am just writing down my thoughts here as an FYI since this subject is indeed very difficult and subjective. I just wanted to share my thoughts so that you know what I would have done if I where to implement these tests.

Reply:
Thanks for sharing your thoughts 🙏. Completely valid arguments 🌟.
Moved in 0d652fd

Reply:
Double approving this PR then @ashiagr ! 😛 

Feel free to merge it anytime you are ready.